crypto/bio/bss_dgram.c

```diff
@@ -1139,29 +1149,36 @@ static int dgram_sctp_read(BIO *b, char *out, int outl)
                 return -1;
 
             /*
-             * Test if socket buffer can handle max record size (2^14 + 2048
-             * + 13)
+             * Test if socket buffer can handle max record size
+             * (SSL3_RT_MAX_PLAIN_LENGTH
+             * + 2048 (where does this value come from?)
+             * + DTLS1_RT_HEADER_LENGTH)
              */
             optlen = (socklen_t) sizeof(int);
             ret = getsockopt(b->num, SOL_SOCKET, SO_RCVBUF, &optval, &optlen);
```

Comment:
In case one wonders why isn't there sizeof(sz.s) != sizeof(sz.i) here. Well, original reason for sz.{s|i} was that there are systems with undefined socklen_t, and sz.{s|i} allows to compile without opting for elaborate #ifdef spaghetti. This subroutine is SCTP one. And question becomes if there is SCTP dev environment without socklen_t. Supposedly no, which means that one can use socklen_t.